### PR TITLE
Hometask6 web

### PR DESCRIPTION
**Pull request by AndriiBakayev**
Finished Hometask6-web.
Extra features:
Added regular expression for avoiding line wraps in postcodes
Added omitted pagination list to write in page width
Added unique letters list update according applied stFilter

### CODE REVIEW

File: src/web/functions.php [hunk 1]
```diff
@@ -1,17 +1,101 @@
...
+    $pageNavigatorWindow = 10; //How many pages to show around $activePage
+    $listSize = max($listSize, 1);
+    $activePage = max($activePage, 1);
+    $numberOfPages = ceil($listSize / $itemsPerPage);
+    $activePage = min($activePage, $numberOfPages);
+
+    //First and last:
+    $navigationPages = array(1 => 1 , $numberOfPages => $numberOfPages);
+    //Every Window's step:
+    for ($i = $pageNavigatorWindow; $i < $numberOfPages; $i += $pageNavigatorWindow)
```
Comment (yderevianko-cartera) [inline src/web/functions.php]: please add { and }

File: src/web/functions.php [hunk 2]
```diff
@@ -1,17 +1,101 @@
...
+        $navigationPages[$i] = $i;
+    //Window nearby the activepage:
+    $navigatorPagesBegin = max(1, min($activePage - $pageNavigatorWindow / 2, $numberOfPages - $pageNavigatorWindow));
+    $navigatorPagesEnd = min($numberOfPages, max($activePage + $pageNavigatorWindow / 2, $pageNavigatorWindow));
+    for ($i = $navigatorPagesBegin; $i <= $navigatorPagesEnd; $i++)
```
Comment (yderevianko-cartera) [inline src/web/functions.php]: please add { and }

File: src/web/functions.php [hunk 3]
```diff
@@ -1,17 +1,101 @@
...
+        $navigationPages[$i] = $i;
+    //Omissions:
+    for ($i = 2; $i < $numberOfPages; $i++) {
+        if (isset($navigationPages[$i])) continue; //navigate to next ommited page
```
Comment (yderevianko-cartera) [inline src/web/functions.php]: please add { and }

File: src/web/functions.php [hunk 4]
```diff
@@ -1,17 +1,101 @@
...
+function getNavigationPages(int $listSize = 1, int $itemsPerPage = 5, int &$activePage = 1): array
+{
+    $pageNavigatorWindow = 10; //How many pages to show around $activePage
+    $listSize = max($listSize, 1);
+    $activePage = max($activePage, 1);
+    $numberOfPages = ceil($listSize / $itemsPerPage);
+    $activePage = min($activePage, $numberOfPages);
+
+    //First and last:
+    $navigationPages = array(1 => 1 , $numberOfPages => $numberOfPages);
```
Comment (yderevianko-cartera) [inline src/web/functions.php:79]: please use [] instaed of array
And why do you need element 1 => 1 ?

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/web/functions.php (hunk 4 above)
Comment (AndriiBakayev) [reply src/web/functions.php:79]: * returns sorted array of Pages ($key=page no, $val=text) which includes:
[1] is a number of page, "1" or "..." is a text of link, "..." if it is an omission
[2] is a number of page, "2" or '...' is a text of link

Comment (AndriiBakayev) [reply src/web/functions.php:79]: изображение

File: src/web/functions.php (hunk 1 above)
Comment (AndriiBakayev) [reply src/web/functions.php]: Done

File: src/web/functions.php (hunk 2 above)
Comment (AndriiBakayev) [reply src/web/functions.php]: Done

File: src/web/functions.php (hunk 3 above)
Comment (AndriiBakayev) [reply src/web/functions.php]: Done

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**